### PR TITLE
feat: update auth documentation for app router

### LINKED ISSUE #1882: feat: NextAuth documentation update for app router
### Is your feature request related to a problem? Please describe.

The [NextAuth](https://create.t3.gg/en/usage/next-auth) documentation describes the use of `<SessionProvider />` that is the recommended practice for versions of Next.js prior to 14. The documentation also doesn't reflect the actual state of -create-t3-app' since it now uses the `getServerSession` hook instead and the page routing alongside the code snippets are not up to date.  

### Describe the solution you'd like to see

An update to the documentations describing the use of `getServerSession` alongside updated code snippets that match the repository and matching explanations.  

### Describe alternate solutions

Provide a user alert at the top of the page stating that the documentation may be outdated and is currently being updated to reflect the actual state of `create-t3-app`

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by akshithio**
Closes #1882

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Update sections about server-side fetching and use of tRPC + add warning about Client Provider section to highlight that it is relevant only to the use of the `/pages` router. Change one outdated hyperlink and add additional context in few areas for user clarity. 

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

**Comment by akshithio** [comment]
i could add an additional commit to this pr to fix #1889 too maybe?

File: www/src/pages/en/usage/next-auth.mdx [hunk 1]
```diff
@@ -39,27 +46,33 @@ const User = () => {
 
 ## Retrieving session server-side
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/next-auth.mdx:47]: maybe this should be split to 2 sections (pages and app)

the old pages setup is still relevant afaik?

Comment (akshithio) [reply www/src/pages/en/usage/next-auth.mdx:47]: I'm not sure i understand? The pages setup is still very relevant but i thought a warning might be a good call. In theory, the server-side retrieval section serves as the "other section" that explains how auth works in app router situations.

Comment (juliusmarminge) [reply www/src/pages/en/usage/next-auth.mdx:47]: you removed the gSSP part of the docs explaining how to retrieve session on the server in pages router.

**Review by juliusmarminge — APPROVED** [review]
Still a bit hesitant about removing the gSSP pattern as there's been some issues reporting invalid usage and these docs sort of fixed that.

You could add a toggle for this session like #1900 maybe?

We should have a more global switch later, but for now having a tab switch in individual places where it makes sense will have to be good enough 😅

**Comment by akshithio** [comment]
did not see that PR before! seems like an easy enough change, I'll do that by tomorrow! Should I also add another commit into this same pr to fix #1889 because it's a similar sort of issue or would you prefer I open a different pr for that after this is closed?

File: www/src/pages/en/usage/next-auth.mdx (hunk 1 above)
Comment (akshithio) [reply www/src/pages/en/usage/next-auth.mdx:47]: I understand what you mean now.

**Comment by juliusmarminge** [comment]
> Should I also add another commit into this same pr to fix https://github.com/t3-oss/create-t3-app/issues/1889 because it's a similar sort of issue or would you prefer I open a different pr for that after this is closed?

Sure!

**Comment by akshithio** [comment]
I think I'd have to add an additional dependency in the form `@mdx-js/react` and `@mdx-js/runtime` since unlike #1900 where there is only one edit at the top of the page, there's both a continuous series of edits for this specific page and a few of them include code snippets so I'd have to pass down some `.mdx` from the component to the mark down file.

**Comment by juliusmarminge** [comment]
you can make a react component that conditionally renders children based on some condition

so

```mdx
<PagesOnly>
  Use gSSP...
</PagesOnly>
<AppOnly>
  Await in an RSC...
</AppOnly>
```

**Comment by juliusmarminge** [comment]
The trpc stuff was fixed in ff32e083970357a64ddf53e2e0469f9170c06412

Feel free to reopen for the auth changes if you get the time for it in the future

**Pull request closed without merging**